### PR TITLE
Use shared pipelines

### PR DESCRIPTION
**Pull request by nvaytet**
- We make use of the shared pipelines in the https://github.com/scipp/pipelines repo to replace current pipelines.
- We are also now building the `ess` package as a single `noarch` package (platform and python version independent).

### CODE REVIEW

File: .azure-pipelines/pull_request.yml [hunk 1]
```diff
@@ -1,3 +1,21 @@
...
+# found on the Azure pipelines web interface under
+# scipp > scippneutron > Settings > Service connections
+
+resources:
+  repositories:
+  - repository: SharedPipelines
+    name: scipp/pipelines
+    type: github
+    endpoint: scipp
+    ref: allow_for_ess_pipeline
```
Comment (OwenArnold) [inline .azure-pipelines/pull_request.yml]: Update this after dependent `pipelines` PR is merged.

File: .azure-pipelines/release.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+extends:
+  template: stages.yml@SharedPipelines
+  parameters:
+    verbose: ${{ parameters.verbose }}
+    conda_channels: '--channel scipp'
+    gh_pages_subdirectory: 'ess'
+    config:
+      noarch:
+        py_versions: ['3.7']
```
Comment (OwenArnold) [inline .azure-pipelines/release.yml:35]: As a note, this should be the minimum supported version we want to support to ensure compatibility.

File: ess-developer-next.yml [hunk 3]
```diff
@@ -6,20 +6,27 @@ channels:
...
 dependencies:
-  - scipp/label/next::scippneutron
-  - mantid-framework
-  - pytest
-  - sphinx>=1.6
-  - sphinx_rtd_theme
-  - tifffile
-  - astropy 
+  # Build
+  - scippneutron
```
Comment (OwenArnold) [inline ess-developer-next.yml]: needs to be `scipp/label/next::scippneutron`

**Comment by nvaytet** [comment]
I've checked that I can install and import the `noarch` packages in a 3.7 and 3.8 environment with env file:
```yaml
name: ess-py38

channels:
  - conda-forge
  - scipp
  - nodefaults

dependencies:
  # Build
  - scippneutron
  - gitpython

  # Run
  - astropy
  - matplotlib
  - tifffile
  - python=3.8
  - ipython
  - jupyter
  - ipywidgets
  - scipy
  - scipp/label/test::ess
```
and `conda env create -f ess-py38.yml`.

@OwenArnold can you check on osx?

**Comment by nvaytet** [comment]
Then i think we are good to go, and remember to merge https://github.com/scipp/pipelines/pull/7 before we merge this one.

**Review by OwenArnold — APPROVED** [review]

**Comment by nvaytet** [comment]
For the record, to get the docs published from the Main pipeline, I had forgotten to change on Azure the name of the ssh secure key file from `azure-ess-key` to `gh_pages_deploy_key`, and to copy the contents of the ssh public key (https://github.com/scipp/ess/blob/86109ec757b07a2c606047ba4971a2d94a05ec99/.azure-pipelines/documentation_deploy.yml#L26) to a new `gh_pages_deploy_key_public` secure file on Azure.

**Pull request merged**